cartography/intel/aws/s3accountpublicaccessblock.py

```diff
@@ -0,0 +1,153 @@
+    """
+    client = boto3_session.client("s3control", region_name=region)
+    try:
+        account_id = boto3_session.client("sts").get_caller_identity()["Account"]
+        response = client.get_public_access_block(AccountId=account_id)
+        return [response]
+    except ClientError as e:
+        if e.response["Error"]["Code"] == "NoSuchPublicAccessBlockConfiguration":
+            logger.warning(
+                f"No public access block configuration found for account in region {region}"
+            )
+            return []
+        else:
+            raise
+
+
+def transform_account_public_access_block(
+    public_access_block: Dict,
+    region: str,
+    aws_account_id: str,
+) -> Dict:
+    """
+    Transform S3 Account Public Access Block data for ingestion.
+    """
+    pab = public_access_block.get("PublicAccessBlockConfiguration", {})
+    return {
+        "id": f"{aws_account_id}:{region}",
+        "account_id": aws_account_id,
+        "region": region,
+        "block_public_acls": pab.get("BlockPublicAcls"),
+        "ignore_public_acls": pab.get("IgnorePublicAcls"),
+        "block_public_policy": pab.get("BlockPublicPolicy"),
+        "restrict_public_buckets": pab.get("RestrictPublicBuckets"),
+    }
+
+
+@timeit
+def load_account_public_access_block(
+    neo4j_session: neo4j.Session,
+    data: List[Dict],
+    region: str,
+    aws_account_id: str,
+    update_tag: int,
+) -> None:
+    """
+    Load S3 Account Public Access Block information into the graph.
+    """
+    logger.info(
+        f"Loading {len(data)} S3 Account Public Access Block configurations for region {region} into graph."
+    )
+
+    load(
+        neo4j_session,
+        S3AccountPublicAccessBlockSchema(),
+        data,
+        lastupdated=update_tag,
+        region=region,
+        AWS_ID=aws_account_id,
+    )
+
+
+@timeit
+def cleanup_account_public_access_block(
+    neo4j_session: neo4j.Session, common_job_parameters: Dict
+) -> None:
+    """
+    Run S3 Account Public Access Block cleanup job.
+    """
+    logger.debug("Running S3 Account Public Access Block cleanup job.")
+    cleanup_job = GraphJob.from_node_schema(
+        S3AccountPublicAccessBlockSchema(), common_job_parameters
+    )
+    cleanup_job.run(neo4j_session)
+
+
+@timeit
+def sync(
+    neo4j_session: neo4j.Session,
+    boto3_session: boto3.session.Session,
+    regions: List[str],
+    current_aws_account_id: str,
+    update_tag: int,
+    common_job_parameters: Dict,
+) -> None:
+    """
+    Sync S3 Account Public Access Block settings for all regions.
+    """
+    for region in regions:
+        logger.info(
+            f"Syncing S3 Account Public Access Block for {region} in account {current_aws_account_id}"
+        )
+
+        public_access_blocks = get_account_public_access_block(boto3_session, region)
+        transformed_data = []
+
+        for public_access_block in public_access_blocks:
```

Comment:
Move this loop into `transform_account_public_access_block()` function so that the `sync_()` function doesn't need to do this looping